app/components/home/transaction-list/transaction-list-item-pending.tsx

```diff
@@ -1,41 +1,53 @@
-import React, { FC } from 'react';
-import { useHover } from 'use-events';
-import { Box, Flex, Text } from '@blockstack/ui';
+import React, { FC, useRef, RefObject, useEffect, MutableRefObject } from 'react';
+import { useHover, useFocus } from 'use-events';
+import { Box, Text } from '@blockstack/ui';
 
-import { PendingTransaction } from '../../../store/pending-transaction';
-import { listHoverProps, EnableBefore } from './transaction-list-item-hover';
+import { PendingTransaction } from '@store/pending-transaction';
+import { toHumanReadableStx } from '@utils/unit-convert';
 import { TransactionIcon } from './transaction-icon';
-import { toHumanReadableStx } from '../../../utils/unit-convert';
+import { TransactionListItemContainer } from './transaction-list-item-container';
 
 interface TransactionListItemPendingProps {
   tx: PendingTransaction;
+  domNodeMapRef: MutableRefObject<any>;
+  activeTxIdRef: MutableRefObject<string | null>;
   onSelectTx: (txId: string) => void;
 }
 
-export const TransactionListItemPending: FC<TransactionListItemPendingProps> = ({
-  tx,
-  onSelectTx,
-}) => {
-  const [hovered, bind] = useHover();
+export const TransactionListItemPending: FC<TransactionListItemPendingProps> = args => {
+  const { tx, domNodeMapRef, activeTxIdRef, onSelectTx } = args;
+  const [hovered, bindHover] = useHover();
+  const [focused, bindFocus] = useFocus();
+  const containerRef = useRef<HTMLButtonElement>(null);
+
+  useEffect(() => {
+    if (containerRef.current !== null && domNodeMapRef !== null) {
+      domNodeMapRef.current[tx.tx_id] = containerRef.current;
+    }
+  }, [domNodeMapRef, tx.tx_id]);
+
+  if (focused && activeTxIdRef !== null) {
+    activeTxIdRef.current = tx.tx_id;
+  }
 
   return (
-    <Flex
-      as={EnableBefore}
-      mb="loose"
-      cursor="pointer"
-      position="relative"
-      _before={listHoverProps(hovered)}
-      onClick={() => onSelectTx(tx.txId)}
-      data-txid={tx.txId}
-      {...bind}
+    <TransactionListItemContainer
+      ref={(containerRef as unknown) as RefObject<HTMLDivElement>}
+      onClick={() => onSelectTx(tx.tx_id)}
+      data-txid={tx.tx_id}
+      focused={focused}
+      hovered={hovered}
+      txId={tx.tx_id}
+      {...bindHover}
+      {...bindFocus}
     >
       <TransactionIcon variant="pending" mr="base-loose" />
       <Box flex={1}>
         <Text textStyle="body.large.medium" display="block">
           Sending
         </Text>
         <Text textStyle="body.small" color="ink.600">
-          {tx.txId.substr(0, 28)}
+          {tx.tx_id.substr(0, 28)}…
```

Comment:
I assume this comes from design guidance, but 28 seems like an odd length to prefix a transaction ID

Reply:
This is an arbitrary number I pulled from the depths of my imagination. The whole ID is too long. Is there are more common truncation point?